core/commands/files/files.go

```diff
@@ -69,6 +73,17 @@ var hashOption = cmdkit.StringOption("hash", "Hash function to use. Will set Cid
 
 var formatError = errors.New("Format was set by multiple options. Only one format option is allowed")
 
+type StatOutput struct {
+	Hash           string
+	Size           uint64
+	CumulativeSize uint64
+	Blocks         int
+	Type           string
+	WithLocality   bool
```

Comment:
Please add `` `json:",omitempty"` `` after WithLocality so that the API output does not change if the `--with-local` option is used.

Reply:
Someone to weight here ? For me it makes more sense like this. There is no compatibility problem, it's just an extra field. The problem is If `WithLocality` is set to false, the field is not present in the api anwser so you need to test if the field is present *and* if it's set to true. Not the end of the world, but ...

Reply:
@MichaelMure No, the same struct would be used on both sides. theres no need to check if its set, we just don't want the API to change where its not wanted. @kevina is right here.